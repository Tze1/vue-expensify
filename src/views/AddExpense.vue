<template>
  <div class="addexpense component">
    <h2 class="addexpense-title">Add Expense</h2>
    <ExpenseForm :submitExpense="addExpense"/>
    <button class="addexpense-cancel button exit" @click="onCancel">Cancel</button>
  </div>
</template>

<script>
  import ExpenseForm from '../components/ExpenseForm';

  export default {
    name: 'AddExpense',
    methods: {
      onCancel () {
        this.$router.push('/dashboard');
      },
      addExpense (expense) {
        this.$store.dispatch('addExpense', expense);
        this.$router.push('/dashboard');
      },
    },
    components: {
      ExpenseForm,
    },
  };
</script>

<style scoped lang="scss">
  @import '../styles/app.scss';
</style>
