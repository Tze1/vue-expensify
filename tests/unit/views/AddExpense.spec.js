import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import AddExpense from '@/views/AddExpense';
import testExpenses from '../../fixtures/testExpenses';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('AddExpense.vue', () => {

  it('renders AddExpense view properly', () => {
    const wrapper = shallowMount(AddExpense);
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches addExpense action properly', () => {
    const dispatchStub = jest.fn();
    const pushStub = jest.fn();
    const $store = {
      state: {},
      dispatch: dispatchStub,
    };
    const $router = {
      push: pushStub,
    };

    const testExpense = {
      createdAt: testExpenses[0].createdAt,
      description: testExpenses[0].description,
      amount: testExpenses[0].amount,
      note: testExpenses[0].note,
    };
    const wrapper = shallowMount(AddExpense, {
      localVue,
      mocks: {
        $store,
        $router,
      },
    });

    wrapper.vm.addExpense(testExpense);

    expect(dispatchStub).toHaveBeenCalledWith('addExpense', testExpense);
    expect(pushStub).toHaveBeenCalledWith('/dashboard');
  });
});
