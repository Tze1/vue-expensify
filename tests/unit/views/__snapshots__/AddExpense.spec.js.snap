// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpense.vue renders AddExpense view properly 1`] = `
<div class="addexpense component">
  <h2 class="addexpense-title">Add Expense</h2>
  <expenseform-stub submitexpense="function () { [native code] }"></expenseform-stub> <button class="addexpense-cancel button exit">Cancel</button>
</div>
`;
